### ISSUE TITLE
Generation of standalone lexer/parser source file

### ISSUE THREAD

**Issue by nshibano** [fsprojects/FsLexYacc#79]
By default, generated lexer/parser .fs file needs external FsLexYacc.Runtime library. But in some situation, standalone .fs file is useful. For example,

- When you don't want to add package dependency to FsLexYacc and/or FsLexYacc.Runtime
- When the .NET version you are targeting is incompatible with FsLexYacc.Runtime's one

Available solution close to standalone .fs file is below.

1. Download .nuget package, rename it to .zip, extract fslex.exe and fsyacc.exe.
2. Using these .exe's, compile .fsl and .fsy with appropriate --lexlib and --parselib options.
3. From source code of FsLexYacc, copy Lexing.fs and Parsing.fs to your project and rename namespaces in it if needed.
4. Compile everything together.

The problem with this solution is it can cause inconsistency between copied runtime and generated lexer/parser files when FsLexYacc version is updated, and re-copying of runtime source will be needed.

**Comment by kkm000** [comment]
Exactly. I am not aware of any existing infrastructure that would let you do that. You can technically do
```
<PackageReference .... PrivateAssets="All" />
```
and then package the dll (and FSharp.Core too!) from the build directory into a single nuget, but I do not see what exactly does it achieve.

**Comment by kkm000** [comment]
Note that you need to manually add `PrivateAssets="All"` to FsLexYacc compilers package always, because otherwise it will be listed as a dependency in the generated NuGet. There is a bug tracking that: https://github.com/NuGet/Home/issues/4125 -- looks like it is marked as fixed, but I have no idea whether it was rolled out. So at the very least, check that the stanza is there.

As for the .Runtime package, I do not really understand what would copying the DLLs achieve, really. Depending on it is a pretty normative practice (it will pull the FSharp.Core transitively, too, and let NuGet unify references in the project using your package. Otherwise, if you package the runtime and Core into your project, you'll likely end with (1) a clash of DLLs copied to its bin directory and (b) no autogenerated binding redirects. So I would advise against that pretty strongly.

Did it answer your question? Can we close this? Or please explain what do you have in mind exactly.
